### PR TITLE
Read-only displays for signature, duration, and time of day

### PR DESCRIPTION
**Pull request by gigxz**
## Description

Add "read-only" form displays for some elements that don't have them yet.
(In the future I will remember to add the read-only displays to the feature specs for new form components !  )

Also includes some minor style changes.

## Type of change
- [x] New feature (adds functionality)

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (eslint)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: src/components/elements/input/MinutesDurationInput.tsx [hunk 1]
```diff
@@ -90,6 +92,8 @@ const MinutesDurationInput = ({
           helperText={null}
           max={59}
           fullWidth={false}
+          inputWidth={105}
+          placeholder='0'
```
Comment (gigxz) [inline src/components/elements/input/MinutesDurationInput.tsx:100]: Some very minor style changes on this component.  Fixed width feels like an anti-pattern but it bothered me how wide the input space was, when we know it's only 2 chars max. I tested this on mobile and it looks fine, so I think its OK.

Screenshot 2024-02-22 at 3 16 10 PM

File: src/modules/form/components/viewable/DynamicViewGroup.tsx [hunk 2]
```diff
@@ -52,6 +57,8 @@ const DynamicViewGroup = (props: ViewGroupItemComponentProps) => {
       );
     case Component.Table:
       return <TableGroup key={props.item.linkId} {...props} viewOnly />;
+    case Component.SignatureGroup:
+      return <SignatureGroupCard key={props.item.linkId} viewOnly {...props} />;
```
Comment (gigxz) [inline src/modules/form/components/viewable/DynamicViewGroup.tsx:61]: Updated `DynamicViewGroup` to support read-only display of signature items:

Screenshot 2024-02-22 at 3 29 27 PM

File: src/modules/form/components/viewable/DynamicViewField.tsx [hunk 3]
```diff
@@ -103,9 +106,27 @@ const DynamicViewField: React.FC<DynamicViewFieldProps> = ({
...
     case ItemType.String:
+      return <TextContent {...commonProps} />;
     case ItemType.Integer:
+      if (item.component === Component.MinutesDuration) {
+        return (
+          <TextContent
+            {...commonProps}
+            renderValue={(val) => {
+              const [hours, minutes] = minutesToHoursAndMinutes(val);
+              return formatDuration({ minutes, hours }, { zero: true });
```
Comment (gigxz) [inline src/modules/form/components/viewable/DynamicViewField.tsx:125]: read-only display for minutes duration

Screenshot 2024-02-22 at 3 49 19 PM

File: src/modules/form/components/viewable/DynamicViewField.tsx [hunk 4]
```diff
@@ -103,9 +106,27 @@ const DynamicViewField: React.FC<DynamicViewFieldProps> = ({
         />
       );
     case ItemType.TimeOfDay:
+      return (
+        <TextContent
+          {...commonProps}
+          renderValue={(val) => formatTimeOfDay(val)}
```
Comment (gigxz) [inline src/modules/form/components/viewable/DynamicViewField.tsx:112]: Read-only display for time of day

Screenshot 2024-02-22 at 3 44 20 PM

File: src/modules/form/components/group/SignatureGroupCard.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { GroupItemComponentProps } from '../../types';
+import FormCard, { FormCardProps } from './FormCard';
+import { SignatureIcon } from '@/components/elements/SemanticIcons';
+
+/**
+ * Render a top-level form card with Signature Icon and special styling.
+ * Usually has Signature group item(s) as children.
+ */
+const SignatureGroupCard = ({
```
Comment (gigxz) [inline src/modules/form/components/group/SignatureGroupCard.tsx:9]: refactored into its own component since it is now used by both DynamicGroup and DynamicViewGroup

File: src/modules/form/components/group/Signature.tsx [hunk 6]
```diff
@@ -21,25 +27,25 @@ const Signature = ({ item, renderChildItem }: GroupItemComponentProps) => {
...
       <fieldset style={{ border: 'none', margin: 'none', padding: 'none' }}>
-        {item.text && (
-          <legend>
-            <Typography
-              variant='h5'
-              color={theme.palette.primary.main}
-              sx={{ mb: 2 }}
-            >
-              {item.text}
+        <Stack gap={2}>
```
Comment (gigxz) [inline src/modules/form/components/group/Signature.tsx:30]: Added stack since certification should only be shown if `viewOnly==false`, and we dont want to keep the extra padding below the children. This is why I usually go for Stack rather than adding `sx={{ mb: 2 }}` between components

File: src/components/elements/input/NumberInput.tsx [hunk 7]
```diff
@@ -85,7 +85,7 @@ const NumberInput = ({
 
   return (
     <TextInput
-      type='number'
+      type='text'
```
Comment (gigxz) [inline src/components/elements/input/NumberInput.tsx:88]: turns off up/down arrow keys on the input field. It still functions as numeric from the input mode

File: src/components/elements/input/MinutesDurationInput.tsx [hunk 8]
```diff
@@ -9,6 +9,10 @@ type Props = { value?: number; onChange?: (val: number | null) => void } & Omit<
   'value' | 'onChange'
 >;
 
+export const minutesToHoursAndMinutes = (minutes: number) => {
+  return [Math.floor(minutes / 60), minutes % 60];
```
Comment (ttoomey) [inline src/components/elements/input/MinutesDurationInput.tsx]: maybe more accurate to use `as const` and avoid type widening? 
```suggestion
  return [Math.floor(minutes / 60), minutes % 60] as const;
```

File: src/modules/form/components/group/Signature.tsx [hunk 9]
```diff
@@ -21,25 +27,25 @@ const Signature = ({ item, renderChildItem }: GroupItemComponentProps) => {
...
+          {item.text && (
+            <legend>
+              <Typography variant='h5' color={theme.palette.primary.main}>
+                {item.text}
+              </Typography>
+            </legend>
```
Comment (ttoomey) [inline src/modules/form/components/group/Signature.tsx]: one less element?
```suggestion
            <Typography variant='h5' color={theme.palette.primary.main} component='legend'>
              {item.text}
            </Typography>
```

File: src/modules/form/components/group/Signature.tsx (hunk 6 above)
Comment (martha) [reply src/modules/form/components/group/Signature.tsx:30]: great!

File: src/modules/form/components/group/Signature.tsx [hunk 10]
```diff
@@ -21,25 +27,25 @@ const Signature = ({ item, renderChildItem }: GroupItemComponentProps) => {
...
+          )}
+          <Grid container direction='row' columnGap={viewOnly ? 6 : 2}>
+            {renderChildItem &&
+              item.item?.map((childItem) => renderChildItem(childItem))}
+          </Grid>
+          {!viewOnly && (
+            <Typography variant='body2'>
+              By signing, I do hereby certify that the above information is
+              true, accurate and complete to the best of my knowledge
```
Comment (ttoomey) [inline src/modules/form/components/group/Signature.tsx:47]: This is a bit of scope creep but I think this language isn't quite right. Folks aren't always signing in this context. They are also signing to allow release of information. Maybe:
"Entering my name and selecting this box constitutes my electronic signature and agreement to be legally bound"

File: src/test/utils/testUtils.ts [hunk 11]
```diff
@@ -15,6 +15,8 @@ const mockValueForItem = (item: FormItem) => {
       return sample([0, 50.0, 100.25]);
     case ItemType.Date:
       return new Date();
+    case ItemType.TimeOfDay:
+      return (60 * 6 + 30).toString(); // 6:30PM (minutes since midnight)
```
Comment (martha) [inline src/test/utils/testUtils.ts]: if it's minutes since midnight then this is 6:30 AM not PM, right?

**Review by martha — APPROVED** [review]
All lovely thank you @gigxz ‼️

**Review by ttoomey — APPROVED** [review]
Looks great!

File: src/modules/form/components/group/Signature.tsx (hunk 10 above)
Comment (gigxz) [reply src/modules/form/components/group/Signature.tsx:47]: agree completely, I had a similar thought. moved to PT https://www.pivotaltracker.com/story/show/187100953
